contracts/test/_metastrategies-fixtures.js

```diff
@@ -0,0 +1,221 @@
+
+  await balanceMetaPool(fixture);
+
+  return fixture;
+}
+
+async function balanceMetaPool(fixture) {
+  const { vault, domen, ousdMetaPool } = fixture;
+
+  // Balance metapool
+  const ousdBalance = await ousdMetaPool.balances(0);
+  const crv3Balance = await ousdMetaPool.balances(1);
+
+  const exchangeSign = "exchange(int128,int128,uint256,uint256)";
+  const exchagneMethod = await ousdMetaPool.connect(domen)[exchangeSign];
+  if (ousdBalance.gt(crv3Balance)) {
+    // Tilt to 3CRV
+    await exchagneMethod(1, 0, ousdBalance.sub(crv3Balance).div(2), 0);
+  } else if (crv3Balance.gt(ousdBalance)) {
+    // Tilt to OUSD
+    await exchagneMethod(0, 1, crv3Balance.sub(ousdBalance).div(2), 0);
+  }
+
+  await vault.connect(domen).allocate();
+  await vault.connect(domen).rebase();
+}
+
+async function withCRV3TitledMetapool() {
+  const fixture = await loadFixture(withBalancedMetaPool);
+
+  await tiltTo3CRV(fixture);
+
+  return fixture;
+}
+
+async function tiltTo3CRV(fixture, amount) {
+  const { vault, domen, ousdMetaPool } = fixture;
+
+  amount = amount || ousdUnits("500000");
+
+  // Tilt to 3CRV by half a million
+  const exchangeSign = "exchange(int128,int128,uint256,uint256)";
+  await ousdMetaPool.connect(domen)[exchangeSign](1, 0, amount, 0);
+
+  await vault.connect(domen).allocate();
+  await vault.connect(domen).rebase();
+}
+
+async function withOUSDTitledMetapool() {
+  const fixture = await loadFixture(withBalancedMetaPool);
+
+  await tiltToOUSD(fixture);
+
+  return fixture;
+}
+
+async function tiltToOUSD(fixture, amount) {
+  const { vault, domen, ousdMetaPool } = fixture;
+
+  amount = amount || ousdUnits("500000");
+
+  // Tilt to 3CRV by half a million
+  const exchangeSign = "exchange(int128,int128,uint256,uint256)";
+  await ousdMetaPool.connect(domen)[exchangeSign](0, 1, amount, 0);
+
+  await vault.connect(domen).allocate();
+  await vault.connect(domen).rebase();
+}
+
+async function get3CRVLiquidity(fixture, crv3Amount) {
+  const { threepoolSwap } = fixture;
+  const vPrice = await threepoolSwap.get_virtual_price();
+  return crv3Amount.div(vPrice).mul(ousdUnits("1"));
+}
+
+async function addLiquidity(fixture, ousdAmount, crv3Liquidity, receiver) {
+  const { domen, ousdMetaPool } = fixture;
+
+  receiver = receiver ? receiver.getAddress() : domen.getAddress();
+
+  const addSignature = "add_liquidity(uint256[2],uint256,address)";
+  const addMethod = await ousdMetaPool.connect(domen)[addSignature];
+  addMethod([ousdAmount, crv3Liquidity], 0, receiver);
+}
+
+async function removeAllLiquidity(fixture, user) {
+  const { ousdMetaPool } = fixture;
+
+  const address = user.getAddress();
+  const removeSign = "remove_liquidity(uint256,uint256[2])";
+  const removeMethod = await ousdMetaPool.connect(user)[removeSign];
+  removeMethod(ousdMetaPool.balanceOf(address), ["0", "0"]);
+}
+
+async function removeLiquidityImbalanced(fixture, user, crv3Liquidity) {
```

Comment:
Maybe add comment: 
// removing liquidity in a manner where we first remove required amount of 3crv. And with whatever LP remains we remove OUSD